skyportal/handlers/api/analysis.py

```diff
@@ -424,3 +467,510 @@ def delete(self, analysis_service_id):
+                status='queued',
+                handled_by_url="/webhook/obj_analysis",
+                invalid_after=invalid_after,
+            )
+        # Add more analysis_resource_types here one day (eg. GCN)
+        else:
+            return self.error(
+                f'analysis_resource_type must be one of {", ".join(["obj"])}',
+                status=404,
+            )
+
+        DBSession().add(analysis)
+        try:
+            self.verify_and_commit()
+        except IntegrityError as e:
+            return self.error(f'Analysis already exists: {str(e)}')
+        except Exception as e:
+            return self.error(f'Unexpected error creating analysis: {str(e)}')
+
+        # Now call the analysis service to start the analysis, using the `input` data
+        # that we assembled above.
+        external_analysis_service = functools.partial(
+            self.call_external_analysis_service,
+            analysis_service.url,
+            f'{get_app_base_url()}/{analysis.handled_by_url}/{analysis.token}',
+            inputs=inputs,
+            authentication_type=analysis_service.authentication_type,
+            authinfo=analysis_service.authinfo,
+            callback_method="POST",
+            invalid_after=invalid_after,
+            analysis_resource_type=analysis_resource_type,
+            resource_id=resource_id,
+        )
+
+        self.push_notification(
+            'Sending data to analysis service to start the analysis.',
+        )
+
+        def analysis_done_callback(
+            future,
+            logger=log,
+            analysis_id=analysis.id,
+            analysis_service_id=analysis_service.id,
+            analysis_resource_type=analysis_resource_type,
+        ):
+            """
+            Callback function for when the analysis service is done.
+            Updates the Analysis object with the results/errors.
+            """
+            # grab the analysis (only Obj for now)
+            if analysis_resource_type.lower() == 'obj':
+                try:
+                    session = Session()
+                    analysis = session.query(ObjAnalysis).get(analysis_id)
+                except Exception as e:
+                    log(f'Could not access Analysis {analysis_id} {e}.')
+                    return
+            else:
+                log(f"Invalid analysis_resource_type: {analysis_resource_type}")
+                return
+
+            analysis.last_activity = datetime.datetime.utcnow()
+            try:
+                result = future.result()
+                analysis.status = 'pending' if result.status_code == 200 else 'failure'
+                # truncate the return just so we dont have a huge string in the database
+                analysis.status_message = result.text[:1024]
+            except Exception:
+                analysis.status = 'failure'
+                analysis.status_message = str(future.exception())[:1024]
```

Comment:
Does this work if the strings are much shorter?

Reply:
```python
$ "yes it does"[:1024]
'yes it does'
```